### PR TITLE
fix: update contributing guide

### PR DESCRIPTION
**Pull request by pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

Partial completion of #887

View changes: [/docs/general/contributor-guide/](https://deploy-preview-888--apache-apisix.netlify.app/docs/general/contributor-guide/)
 
Changes:

1. Fixed typos
2. Added hyperlinks
3. Improved readability by restructuring sentences

### CODE REVIEW

File: website/docs/general/contributor-guide.md
Comment (avinal) [inline website/docs/general/contributor-guide.md:18]: Only `discussions` or `discussion forums` will be better choice of words. Not a major issue though. May leave as it is.

File: website/docs/general/contributor-guide.md
Comment (avinal) [inline website/docs/general/contributor-guide.md:20]: This step can be combined with the last paragraph.

File: website/docs/general/contributor-guide.md
Comment (avinal) [inline website/docs/general/contributor-guide.md:67]: If we are using forked repo then the upstream will be set already. It would be good to write about check and add.

File: website/docs/general/contributor-guide.md
Comment (avinal) [inline website/docs/general/contributor-guide.md]:
```suggestion
**Note**: Setup your name and email address to make sure that your ID shows up in the contributor list.
```

File: website/docs/general/contributor-guide.md
Comment (avinal) [inline website/docs/general/contributor-guide.md]:
```suggestion
git config --global user.name "full name"
git config --global user.email "mail address"
```
It is a bad practice to use the GitHub username in the git config. You should always use your full name and the mail address tied to your GitHub account or any other service you might be using. It is the email that connects the commits to a contributor, not the username.

File: website/docs/general/contributor-guide.md
Comment (avinal) [inline website/docs/general/contributor-guide.md]: Direct merge can introduce extra merge commits in your pull request if it cannot be fast-forwarded. A better and preferred way is to always rebase the pull request branch against the `upstream/main`. If not rebased successfully it will show conflicts. Solve the conflicts and then continue with rebase. This method also ensures less work for the developer merging the PR.

File: website/docs/general/contributor-guide.md
Comment (avinal) [inline website/docs/general/contributor-guide.md]: Please change as suggested above, do ask if you need help.

File: website/docs/general/contributor-guide.md
Comment (pottekkat) [reply website/docs/general/contributor-guide.md:18]: I used "discussions forum" as GitHub calls it "Discussions" and to make it clear that we mean the forum.

File: website/docs/general/contributor-guide.md
Comment (pottekkat) [reply website/docs/general/contributor-guide.md:20]: My thought was to add this as a step to force that it is intended to be followed. It was a step previously and I thought it would be better to not change it. What do you think?

File: website/docs/general/contributor-guide.md
Comment (pottekkat) [reply website/docs/general/contributor-guide.md]: I agree. My usual flow is that before I create a feature branch, I fetch the upstream/master and merge it with my local master and push it to origin. Once my local master and upstream master is same, I create a feature branch. Then, while working on the feature branch, I rebase it with upstream/master to make sure that there aren't any conflicts and I am working on top of the latest changes.

**Comment by pottekkat** [comment]
@avinal Made some of the suggested changes and made some comments. Please check it out.

File: website/docs/general/contributor-guide.md
Comment (avinal) [reply website/docs/general/contributor-guide.md:20]: Ok, now it seems good to me :)

File: website/docs/general/contributor-guide.md
Comment (avinal) [reply website/docs/general/contributor-guide.md]: Yeah, that is nice too, we just have to make sure not to introduce unnecessary commits, whatever serves that is fine.

**Review by avinal — DISMISSED** [review]
Looks great to me 😃 

Cc: @juzhiyuan

**Review by juzhiyuan — DISMISSED** [review]
LGTM

File: website/docs/general/contributor-guide.md
Comment (yzeng25) [inline website/docs/general/contributor-guide.md]: The in-page link works when the link fits the section title. Would be nice to update both of them.
```suggestion
Please follow this guide to [report bugs](#submitting-an-issue), [submit suggestions](#submitting-an-issue), or [submit PRs](#open-a-pull-request).
```

File: website/docs/general/contributor-guide.md
Comment (yzeng25) [inline website/docs/general/contributor-guide.md]: I asked other contributors to mock the process of opening an issue. It turns out that contributors are not able to change/select the labels by their own. I would suggest to delete this step for now.

We would definitely need to figure out the grainarities, accesses, and responsibilities here.
```suggestion

```

File: website/docs/general/contributor-guide.md
Comment (yzeng25) [inline website/docs/general/contributor-guide.md]:
```suggestion
1. Once you have chosen an issue to work on or [opened a new issue](#submitting-an-issue), please comment under the issue and ask a [Committer or PMC](/team) to assign it to you. Or you can claim the issue when you create it.
```

File: website/docs/general/contributor-guide.md
Comment (yzeng25) [inline website/docs/general/contributor-guide.md]: Committer or PMC would be more accurate IMHO.
```suggestion
3. Connect with a [Committer or PMC](/team) for providing feedback on your issue and reviewing your PR.
```

File: website/docs/general/contributor-guide.md
Comment (pottekkat) [reply website/docs/general/contributor-guide.md]: Aaah, the AI autocomplete messed up! Will make these changes.

File: website/docs/general/contributor-guide.md
Comment (pottekkat) [reply website/docs/general/contributor-guide.md]: Yes. I also thought so. Thank you for clarifying. Ultimately, we would want to have multiple templates and the templates should handle the labelling with minimal manual labelling from the maintainers.

**Comment by pottekkat** [comment]
@juzhiyuan @yzeng25 @avinal I have made changes as suggested and resolved the merge conflicts. Please review.

**Review by yzeng25 — APPROVED** [review]
LGTM

**Comment by yzeng25** [comment]
> @juzhiyuan @yzeng25 @avinal I have made changes as suggested and resolved the merge conflicts. Please review.

@juzhiyuan we need one more approval to get this merged.

**Review by avinal — APPROVED** [review]
Everything seems fine.

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**
